File: service/SurfWebAutofac/SurfWebAutofac/Controllers/NewRecordController.cs
using IServices.Main;
using Microsoft.AspNetCore.Mvc;
using Model.Dtos.NewRecords;
using Model.Models.Main;

namespace SurfWebAutofac.Controllers;

/// <summary>
///     新纪录控制器
/// </summary>
[ApiController]
[Route("[controller]")]
public class NewRecordController : ControllerBase
{
    private readonly IMapServices _mapServices;

    /// <summary>
    /// </summary>
    public NewRecordController(IMapServices mapServices)
    {
        _mapServices = mapServices;
    }

    /// <summary>
    ///     获取最新纪录
    /// </summary>
    [HttpGet("GetNewRecordList")]
    public async Task<List<NewRecordDto>> GetNewRecordList(RecordTypeEnum recordType)
    {
        return await _mapServices.GetNewRecordList(recordType);
    }

    /// <summary>
    ///     获取新增地图
    /// </summary>
    [HttpGet("GetNewMapList")]
    public async Task<List<NewMapDto>> GetNewMapList()
    {
        return await _mapServices.GetNewMapList();
    }

    /// <summary>
    ///     获取热门地图
    /// </summary>
    [HttpGet("GetPopularMapList")]
    public async Task<List<PopularMapDto>> GetPopularMapList()
    {
        return await _mapServices.GetPopularMapList();
    }
}

File: service/Java/SurfWeb/SurfWeb/webapi/src/main/java/com/surfweb/web/controller/NewRecordController.java
package com.surfweb.web.controller;

import com.surfweb.common.core.controller.BaseController;
import com.surfweb.system.dto.newrecords.NewMapDto;
import com.surfweb.system.dto.newrecords.NewRecordDto;
import com.surfweb.system.dto.newrecords.PopularMapDto;
import com.surfweb.common.enums.RecordTypeEnum;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import com.surfweb.system.service.IMapServices;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

/**
 * 新纪录
 *
 * @author ruoyi
 */
@RestController
@RequestMapping("/NewRecord")
public class NewRecordController extends BaseController {

    @Autowired
    private IMapServices mapServices;
    /**
     * 获取最新纪录
     */
    @GetMapping("/getNewRecordList")
    public List<NewRecordDto> getNewRecordList(@RequestParam("recordType") RecordTypeEnum recordType) {
        return mapServices.getNewRecordList(recordType);
    }

    /**
     * 获取新增地图
     */
    @GetMapping("/getNewMapList")
    public List<NewMapDto> getNewMapList() {
        return mapServices.getNewMapList();
    }

    /**
     * 获取热门地图
     */
    @GetMapping("/getPopularMapList")
    public List<PopularMapDto> getPopularMapList() {
        return mapServices.getPopularMapList();
    }
}
